packages/@react-native-windows/codegen/src/generators/GenerateTypeScript.ts

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ * @format
+ */
+
+'use strict';
+
+import {
+  NamedShape,
+  NativeModuleBaseTypeAnnotation,
+  NativeModuleFunctionTypeAnnotation,
+  NativeModuleObjectTypeAnnotation,
+  NativeModuleParamTypeAnnotation,
+  NativeModuleReturnTypeAnnotation,
+  NativeModuleSchema,
+  Nullable,
+  SchemaType,
+} from 'react-native-tscodegen';
+
+type ObjectProp = NamedShape<Nullable<NativeModuleBaseTypeAnnotation>>;
+type FunctionParam = NamedShape<Nullable<NativeModuleParamTypeAnnotation>>;
+type FunctionDecl = NamedShape<Nullable<NativeModuleFunctionTypeAnnotation>>;
+type FilesOutput = Map<string, string>;
+
+const moduleTemplate = `
+/*
+ * This file is auto-generated from a NativeModule spec file in js.
+ *
+ * This is a TypeScript turbo module definition file.
+ */
+
+// the following import statements are not actually working today
+import {TurboModule, RootTag} from 'react-native/Libraries/TurboModule/RCTExport';
+import * as TurboModuleRegistry from 'react-native/Libraries/TurboModule/TurboModuleRegistry';
+import {Int32, Float, Double} from 'react-native/Libraries/Types/CodegenTypes';
+'use strict';
+
+::_MODULE_ALIASED_STRUCTS_::
+export interface Spec extends TurboModule {
+::_MODULE_MEMBERS_::
+}
+
+export default TurboModuleRegistry.getEnforcing<Spec>('::_MODULE_NAME_::');
+`;
+
+function translateType(
+  type: Nullable<
+    | NativeModuleBaseTypeAnnotation
+    | NativeModuleParamTypeAnnotation
+    | NativeModuleReturnTypeAnnotation
+  >,
+): string {
+  // avoid: Property 'type' does not exist on type 'never'
+  const returnType = type.type;
+  switch (type.type) {
+    case 'StringTypeAnnotation':
+      return 'string';
+    case 'NumberTypeAnnotation':
+      return 'number';
+    case 'FloatTypeAnnotation':
+      return 'Float';
+    case 'DoubleTypeAnnotation':
+      return 'Double';
+    case 'Int32TypeAnnotation':
+      return 'Int32';
+    case 'BooleanTypeAnnotation':
+      return 'boolean';
+    case 'ArrayTypeAnnotation':
+      if (type.elementType) {
+        return `${translateType(type.elementType)}[]>`;
```

Comment:
the trailing > at the end doesn't have a matching opening <, is that intentional?

Reply:
fixed